contracts/Joint.sol

```diff
@@ -22,123 +28,129 @@ interface IERC20Extended {
     function symbol() external view returns (string memory);
 }
 
+interface ProviderStrategy {
+    function vault() external view returns (VaultAPI);
+
+    function strategist() external view returns (address);
+
+    function keeper() external view returns (address);
+
+    function want() external view returns (address);
+}
+
 contract Joint {
     using SafeERC20 for IERC20;
     using Address for address;
     using SafeMath for uint256;
 
+    ProviderStrategy public providerA;
+    ProviderStrategy public providerB;
+
     address public tokenA;
```

Comment:
can we remove this var and use `providerA.token()`?

Reply:
Might be slightly more expensive gas-wise, but i'm not sure ...